gramex/apps/admin2/gramexadmin.py

```diff
@@ -43,13 +44,35 @@ def get_auth_conf(kwargs):
         # For DBAuth, hoist the user.column into as the id: for the URL
         user_column = auth_kwargs.get('user', {}).get('column', 'user')
         data_conf = gramex.handlers.DBAuth.clear_special_keys(
-            auth_kwargs.copy(), 'user', 'password', 'forgot', 'signup', 'template', 'delay')
+            auth_kwargs.copy(), 'rules', 'user', 'password', 'forgot',
+            'signup', 'template', 'delay')
         data_conf['id'] = user_column
         return authhandler, auth_conf, data_conf
     else:
         raise ValueError('Missing lookup: in url.%s (authhandler)' % authhandler)
 
 
+@coroutine
+def send_welcome_email(handler):
+    if handler.request.method == 'POST':
+        to = handler.get_arg('email', False)
+        if not to:
+            app_log.warning('No email address found for new user {handler.get_arg("user")}.')
+            return
+        email = handler.auth_conf.kwargs.get('forgot', False)
+        if not email:
+            app_log.warning('No email sender found in config.')
+            return
+        mailer = gramex.service.email[email.email_from]
+        yield gramex.service.threadpool.submit(
+            mailer.mail,
+            to=to, subject='Welcome to your Gramex app!',
+            body=dedent(f'''
+                Hello {handler.get_arg('user')},
+                You have been signed up with password {handler.get_arg('user')}.''')
+        )
```

Comment:
Use the authhandler's email_text and email_subject instead. See DBAuth for what they default to

Reply:
- DBAuth signup config accepts email_as, email_from, email_subject, email_text, etc -- same as forgot password
- DBAuth.forgot_password requires a set of kwargs explicitly passed to it, e.g. `self.forgot_password(**self.forgot)` or `self.forgot_password(**self.signup)`
- Admin will send the email using the email_subject, email_text etc under signup, else use the default

Reply:
Ignore this. Just use admin_kwargs for this instead.